### PR TITLE
feat: make KafkaSource configurable

### PR DESCRIPTION
**Pull request by prakharmathur82**
Make Kafka Source configurable to use either the new Kafka Source or the old KafkaConsumer APIs

### CODE REVIEW

File: dagger-core/src/main/java/io/odpf/dagger/core/StreamManager.java [hunk 1]
```diff
@@ -199,6 +197,10 @@ protected StreamInfo createStreamInfo(Table table) {
         return new StreamInfo(stream, table.getSchema().getFieldNames());
     }
 
+    List<Stream> getStreams() {
```
Comment (lavkesh) [inline dagger-core/src/main/java/io/odpf/dagger/core/StreamManager.java:200]: any access modifier?

File: dagger-core/src/main/java/io/odpf/dagger/core/source/kafka/DaggerKafkaSource.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public boolean canHandle() {
+        return SourceName.valueOf(streamConfig.getSourceType()).equals(SourceName.KAFKA_SOURCE);
+    }
+
+    @Override
+    public DataStreamSource register(StreamExecutionEnvironment executionEnvironment, WatermarkStrategy<Row> watermarkStrategy, String streamName) {
+        return executionEnvironment.fromSource(kafkaSource, watermarkStrategy, streamName);
+    }
+
+    KafkaSource getKafkaSource(StreamConfig config, KafkaDeserializationSchema deserializationSchema, Configuration configuration) {
```
Comment (lavkesh) [inline dagger-core/src/main/java/io/odpf/dagger/core/source/kafka/DaggerKafkaSource.java:34]: not sure if the default access is intended here.

File: dagger-core/src/main/java/io/odpf/dagger/core/StreamManager.java (hunk 1 above)
Comment (prakharmathur82) [reply dagger-core/src/main/java/io/odpf/dagger/core/StreamManager.java:200]: it is protected, need this for mocking in tests

File: dagger-core/src/main/java/io/odpf/dagger/core/source/kafka/DaggerKafkaSource.java (hunk 2 above)
Comment (prakharmathur82) [reply dagger-core/src/main/java/io/odpf/dagger/core/source/kafka/DaggerKafkaSource.java:34]: it is protected, need this for mocking in tests

File: dagger-core/src/test/java/io/odpf/dagger/core/StreamManagerTest.java [hunk 3]
```diff
@@ -148,13 +160,16 @@ public void shouldRegisterSourceWithPreprocessorsWithWaterMarks() {
...
 
         StreamManagerStub streamManagerStub = new StreamManagerStub(configuration, env, tableEnvironment, new StreamInfo(dataStream, new String[]{}));
         streamManagerStub.registerConfigs();
         streamManagerStub.registerSourceWithPreProcessors();
 
-        verify(env, Mockito.times(1)).fromSource(any(KafkaSource.class), any(WatermarkStrategy.class), any(String.class));
+        verify(stream, Mockito.times(1)).registerSource(any(StreamExecutionEnvironment.class), any(WatermarkStrategy.class), any(String.class));
     }
 
     @Test
```
Comment (gauravsinghania) [inline dagger-core/src/test/java/io/odpf/dagger/core/StreamManagerTest.java:175]: Add test for idle time config changes

File: version.txt [hunk 4]
```diff
@@ -1 +1 @@
-0.2.5
+0.2.6-rc
```
Comment (gauravsinghania) [inline version.txt:1]: let's change it to 0.2.6

**Pull request merged**